src/System.Runtime.Analyzers/Core/NormalizeStringsToUppercase.cs

```diff
@@ -4,45 +4,98 @@
-                                                                             DiagnosticCategory.Naming,
-                                                                             DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
-                                                                             description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
-                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
         internal static DiagnosticDescriptor ToUpperRule = new DiagnosticDescriptor(RuleId,
                                                                              s_localizableTitle,
                                                                              s_localizableMessageToUpper,
-                                                                             DiagnosticCategory.Naming,
+                                                                             DiagnosticCategory.Globalization,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: @"https://msdn.microsoft.com/en-us/library/bb386042.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(ToUpperInvariantRule, ToUpperRule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(ToUpperRule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
+            analysisContext.RegisterCompilationStartAction(compilationStartContext =>
+            {
+                var stringType = WellKnownTypes.String(compilationStartContext.Compilation);
+                if (stringType == null)
+                {
+                    return;
+                }
+
+                var cultureInfo = compilationStartContext.Compilation.GetTypeByMetadataName("System.Globalization.CultureInfo");
+                var invariantCulture = cultureInfo?.GetMembers("InvariantCulture").OfType<IPropertySymbol>().FirstOrDefault();
```

Comment:
You're right. I was worried since these are source-level analyzers, and the user could pass us any source they wanted. But you're saying that if this symbol doesn't exist, then even if the user _typed_ "`string.ToLower(CultureInfo.InvariantCulture)`" in the source code, it wouldn't resolve to that symbol, because the symbol wouldn't exist, so we can't go down that path. OK, thanks for the explanation.